pkg/controller/authentication/authentication_controller.go

```diff
@@ -168,6 +168,19 @@ func (r *ReconcileAuthentication) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, nil
 	}
 
+	// Make sure the tigera-dex namespace exists, before rendering any objects there.
+	if err := r.client.Get(ctx, client.ObjectKey{Name: render.DexObjectName}, &corev1.Namespace{}); err != nil {
+		if errors.IsNotFound(err) {
+			log.Error(err, "Waiting for namespace tigera-dex to be created")
+			r.status.SetDegraded("Waiting for namespace tigera-dex to be created", err.Error())
+			return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+		} else {
+			log.Error(err, "Error querying tigera-dex namespace")
```

Comment:
Nit, but can you change this to "Failed to retrieve the tigera-dex namespace"?

Reply:
I'd rather leave it like this. Every other resource that is queried/retrieved this way has the exact same message